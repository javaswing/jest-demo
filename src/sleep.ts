export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));

export const timer = (fn: () => void) => {
    setTimeout(fn, 3000);
}