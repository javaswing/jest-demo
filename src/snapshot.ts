export const config1 = () => {
  return {
    method: "GET",
    url: "/api",
    time: new Date(),
  };
};

export const config2 = () => {
  return {
    method: "GET",
    url: "/api",
    time: new Date().getTime(),
  };
};
