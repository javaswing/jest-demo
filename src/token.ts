export const TOKEN_KEY = "token-key";

export function setToken(v?: string) {
  if (v) {
    localStorage.setItem(TOKEN_KEY, v);
  }
}

export function getToken() {
  return localStorage.getItem(TOKEN_KEY);
}
