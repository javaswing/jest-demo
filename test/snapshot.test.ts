import { config1, config2 } from "../src/snapshot";

describe("snapshot test", () => {
  test("test config1 method", () => {
    // 使用 toMatchSnapshot 方法生成快照
    // 会在同级目录下生成 `_snapshots_` 目录
    // 快照根据test的描述，生成对应的映射关系
    // 默认情况下：如果返回的值与快照不同，需要更新快照，否则会报错
    // 在本方法中，因为返回的time是每次都不同的，所以设置time的值为任意的date值
    // 还有一个办法就是，在每次执行后都更新快照 jest --updateSnapshot
    expect(config1()).toMatchSnapshot({
      time: expect.any(Date),
    });
  });

  // 行内快照
  // toMatchInlineSnapshot 方法，将执行快照放到行内中，会放到 toMatchInlineSnapshot 方法中
  test("inline snapshot", () => {
    expect(config2()).toMatchInlineSnapshot(
      {
        time: expect.any(Number),
      },
      `
      {
        "method": "GET",
        "time": Any<Number>,
        "url": "/api",
      }
      `
    );
  });
});
