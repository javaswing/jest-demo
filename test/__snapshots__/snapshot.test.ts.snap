// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test test config1 method 1`] = `
{
  "method": "GET",
  "time": Any<Date>,
  "url": "/api",
}
`;
